### PR TITLE
Zero x trade refactor

### PR DESCRIPTION
**Pull request by nuevoalex**
In order to let the 0x mesh/exchange balanceOf checks function correctly we need to have more data available during that check. In particular the balanceOf check needs to know the:

 - market
 - price
 - outcome
 - orderType

of the order being made.

We do this by:

 - Creating a "ZeroXTrade" token per market
 - Using the ERC1155 interface for this token where a given uint256 token id corresponds to packed:
  - uint240 price
  - uint8 outcome
  - uint8 orderType

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🤘

 This link will always point to this PR's latest changes: https://peek.run/0elhpvz

File: packages/augur-core/source/contracts/trading/IZeroXTrade.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+contract IZeroXTrade {
+
+    struct AugurOrderData {
+        address marketAddress;                  // Market Address
+        uint256 price;                          // Price
+        uint8 outcome;                          // Outcome
+        uint8 orderType;                        // Order Type
+        address kycToken;                       // KYC Token
```
Comment (bthaile) [inline packages/augur-core/source/contracts/trading/IZeroXTrade.sol:15]: no affiliateAddress property needed?

File: packages/augur-artifacts/src/addresses.json [hunk 2]
```diff
@@ -73,24 +73,24 @@
...
-  "CreateOrder": "0x5a21108e505f6c4EB66080a16F078985Ea596FDD",
-  "FillOrder": "0x655caa53Df6e65330A199aa24Adb0C6Dd7F1a7e5",
-  "OICash": "0xC85d65Bd90d34DA3eE7c2D0Ccc378196Fd65c315",
-  "Orders": "0x1Cb97da513c037E2ae4a0b4ae02DC0bc778D3b52",
-  "ProfitLoss": "0xD9A0910a7834370Da626DFF07c5bA3bD6ed3279c",
-  "ShareToken": "0x15983F23c5B977830faf71bCB039F433676Ec045",
-  "SimulateTrade": "0x31C3bFEd7DD39532F074fEd880AFB21cDFF44311",
-  "Trade": "0x59AEb3Ced9B49a7582999babCCC2052CB5914ebF",
-  "ZeroXTradeToken": "0x0280179561827DebC044117774e207D5d1Aab223"
+  "Augur": "0x835f4C94FA66FEc6855998F57dc53C22EDf98afe",
```
Comment (bthaile) [inline packages/augur-artifacts/src/addresses.json]: these look like old kovan addresses.

**Review by bthaile — COMMENTED** [review]
lgtm from what I can understand. 

a few questions.

File: packages/augur-core/source/contracts/trading/IZeroXTrade.sol (hunk 1 above)
Comment (nuevoalex) [reply packages/augur-core/source/contracts/trading/IZeroXTrade.sol:15]: Nope, that's only on fills

File: packages/augur-artifacts/src/addresses.json (hunk 2 above)
Comment (nuevoalex) [reply packages/augur-artifacts/src/addresses.json]: Ah, probably a merge issue then? Will change back

**Review by bthaile — APPROVED** [review]
lgtm

File: packages/augur-core/.soliumrc.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "lbrace": "error",
     "mixedcase": "off",
     "camelcase": "warning",
-    "no-unused-vars": "error",
+    "no-unused-vars": "off",
```
Comment (justinbarry) [inline packages/augur-core/.soliumrc.json:21]: Curious why?

**Review by justinbarry — APPROVED** [review]
LGTM

**Review by bthaile — APPROVED** [review]
Still lgtm

File: packages/augur-sdk/src/state/db/MarketDB.ts [hunk 4]
```diff
@@ -208,33 +229,39 @@ export class MarketDB extends DerivedDB {
...
+      Augur.syncableFlexSearch.addMarketCreatedDocs([log as unknown as MarketCreatedDoc]);
+    }
     log['reportingState'] = MarketReportingState.PreReporting;
     log['invalidFilter'] = false;
     log['marketOI'] = '0x00';
     log['volume'] = '0x00';
     log['disputeRound'] = '0x00';
     log['totalRepStakedInMarket'] = '0x00';
     log['hasRecentlyDepletedLiquidity'] = false;
+    log['liquidity'] = {
```
Comment (justinbarry) [inline packages/augur-sdk/src/state/db/MarketDB.ts:250]: These are defaults yeah?

**Review by justinbarry — APPROVED** [review]
LGTM

File: packages/augur-sdk/src/state/db/MarketDB.ts (hunk 4 above)
Comment (nuevoalex) [reply packages/augur-sdk/src/state/db/MarketDB.ts:250]: yup

**Pull request merged**